### PR TITLE
fix: update code to use federation and gateway plugins

### LINKED ISSUE #117: TypeError: Cannot read properties of undefined
I'm getting the following error on the /federation-schema endpoint on the latest release (0.2.6):
```
TypeError: Cannot read properties of undefined (reading 'serviceMap')
    at Object.<anonymous> (mercurius-federation-info/index.js:21:47
```

### LINKED ISSUE #125: Example server fails to run
Running `npm run example` throws the following error:

```
{"type":"Error","message":"Unknown directive "@key".

Unknown directive "@requires".

Unknown directive "@key".

Unknown directive "@extends".

Unknown directive "@external".

Unknown directive "@external".

Unknown directive "@requires".

Cannot extend type "Mutation" because it is not defined.","stack":"Error: Unknown directive "@key".

Unknown directive "@requires".

Unknown directive "@key".

Unknown directive "@extends".

Unknown directive "@external".

Unknown directive "@external".

Unknown directive "@requires".

Cannot extend type "Mutation" because it is not defined.
    at assertValidSDL (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/graphql/validation/validate.js:135:11)
    at buildASTSchema (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/graphql/utilities/buildASTSchema.js:44:34)
    at buildSchema (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/graphql/utilities/buildASTSchema.js:109:10)
    at fp.name (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/mercurius/index.js:155:14)
    at Plugin.exec (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/avvio/plugin.js:130:19)
    at Boot.loadPlugin (/Users/mathieudavy/dev/nearform/mercurius-federation-info/node_modules/avvio/plugin.js:272:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)"}
```

Node version: 18.14.2
npm: 9.5.0

### PR DESCRIPTION
**Pull request by ekynoxe**
fixes #117 
fixes #125

### CODE REVIEW

**Comment by ekynoxe** [comment]
I probably need to update the README as well.

**Comment by simoneb** [comment]
since the upgrade slipped in without us noticing, do you think there's a way to write a test that would have caught the breakage?

**Comment by simoneb** [comment]
If the example was the only thing affected, then we can get away without the test

**Comment by ekynoxe** [comment]
Yes, the tests were already using @mercurius/gateway: https://github.com/nearform/mercurius-federation-info/blob/master/test/basic.test.js#L42

I've also updated the README to avoid confusion.

**Comment by ekynoxe** [comment]
Actually, @codeflyer @simoneb thinking about it, would we still not have to add a check in https://github.com/nearform/mercurius-federation-info/blob/master/index.js#L21 for `fastify.graphqlGateway` being defined to avoid crashes just like we're seeing in the example server and in other plugins like #117  (which was raised from https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/235)

How can we ensure that with mercurius 12 is used **with** the `@mercurius/gateway` plugin?

**Comment by simoneb** [comment]
I don't remember if there's a reason why we didn't do it, but since this is a fastify plugin and it depends on other plugins to have been registered, shall we not use a plugin dependency then? See https://github.com/fastify/fastify-plugin#dependencies

**Comment by ekynoxe** [comment]
@codeflyer I've added a test and also found another issue with the readme that I've fixed.
I wanted to test the logger output, but being green to `tap` I couldn't find an easy way to mock it and assert a single call to check for the "gateway not present" message (there are other logs that would need to be ignored, and I can't find a way to do what jest would offer with its `toHaveBeenCalledWith` assertion)

I think we should be good to go with this last check. Once merged, we can deploy it and fix https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/235

File: test/basic.test.js [hunk 1]
```diff
@@ -8,6 +9,70 @@ import { readFileSync } from 'fs'
 const fileUrl = new URL('../package.json', import.meta.url)
 const packageJSON = JSON.parse(readFileSync(fileUrl))
 
+test('"/federation-schema" returns a 404 error if fastify is intanciated without the gateway', async t => {
```
Comment (codeflyer) [inline test/basic.test.js:12]: typo: intanciated  -> instantiated

**Comment by ekynoxe** [comment]
Thanks @codeflyer I've implemented the first solution to check the log and fixed the typo.

File: test/basic.test.js [hunk 2]
```diff
@@ -8,6 +9,81 @@ import { readFileSync } from 'fs'
 const fileUrl = new URL('../package.json', import.meta.url)
 const packageJSON = JSON.parse(readFileSync(fileUrl))
 
+test('"/federation-schema" returns a 404 error if fastify is intantiated without the gateway, and logs an error message', async t => {
```
Comment (codeflyer) [inline test/basic.test.js:12]: still missing the `s` -> inStantiated

Comment (ekynoxe) [reply test/basic.test.js]: Argh, sorry about that. 

Fixed (and I've installed a spell checker too!)

**Review by codeflyer — APPROVED** [review]
LGTM

**Pull request merged**